drivers/watchdog/wdt_sam0.c

```diff
@@ -33,9 +34,13 @@ static u32_t wdt_sam0_timeout_to_wdt_period(u32_t timeout_ms)
 {
 	u32_t next_pow2;
 	u32_t cycles;
+	u32_t clk_freq;
 
-	/* Calculate number of clock cycles @ 1.024 kHz input clock */
-	cycles = (timeout_ms * 1024U) / 1000;
+	clock_control_get_rate(device_get_binding(DT_ATMEL_SAM0_WATCHDOG_0_CLOCK_CONTROLLER),
```

Comment:
beware of the 80 chars limit

Reply:
Fixed the cases where I could (by splitting the clock control device into a separate variable and/or moving the initialization out of the definition).  There's a few left that I can't without also having misleading indenting (mostly due to long HAL names combined with existing indents).